### PR TITLE
ci: extend labeling for PRs by change

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding a few more labels for better navigation

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18708.org.readthedocs.build/en/18708/

cc @borda @carmocca

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: .github/label-change.yml [hunk 1]
```diff
@@ -31,3 +31,28 @@ ci:
   - ".github/**"
   - "dockers/**"
   - ".pre-commit-config.yml"
+
+docs:
+  - ".github/workflows/docs-build.yml"
+  - "docs/**"
```
Comment (carmocca) [inline .github/label-change.yml:37]: I would not add the docs label to all PR that touch this path, as many PRs make slight modifications to the docs but they are not generally considered a "docs PR"

```suggestion
```

File: .github/label-change.yml [hunk 2]
```diff
@@ -31,3 +31,28 @@ ci:
...
+  - "requirements/docs.txt"
+  - "requirements/*/docs.txt"
```
Comment (carmocca) [inline .github/label-change.yml:39]: It's more appropiate that a PR that touches requirements is considered  a "dependencies PR". For instance, a dependabot PR is not a "docs PR"

```suggestion
```

File: .github/label-change.yml [hunk 3]
```diff
@@ -31,3 +31,28 @@ ci:
   - ".github/**"
   - "dockers/**"
   - ".pre-commit-config.yml"
+
+docs:
+  - ".github/workflows/docs-build.yml"
```
Comment (carmocca) [inline .github/label-change.yml:36]: Touching CI files is not generally about docs

```suggestion
```

File: .github/label-change.yml
Comment (carmocca) [inline .github/label-change.yml:1]: Remember to grep the repo whenever you rename files. The old filename is referenced in the CI docs

File: .github/label-change.yml (hunk 1 above)
Comment (Borda) [reply .github/label-change.yml:37]: but still, yu may need to check the final docs, the same way you can say about app that a minor style change can add the label without functional change

File: .github/label-change.yml (hunk 3 above)
Comment (Borda) [reply .github/label-change.yml:36]: but it is affecting docs

File: .github/label-change.yml
Comment (Borda) [reply .github/label-change.yml:1]: that is correct as this rename may break docs so shall be checked too

**Review by justusschock — APPROVED** [review]

File: .github/label-change.yml
Comment (carmocca) [reply .github/label-change.yml:1]:
```
$ grep -iIrn ".github/labeler.yml"
.github/workflows/README.md:78:| .github/workflows/labeler-pr.yml, .github/labeler.yml                  | Integration of https://github.com/actions/labeler                                                                                                        |
```

:rabbit:

**Pull request merged**